packages/docusaurus-plugin-content-blog/src/index.ts

```diff
@@ -28,38 +30,13 @@ import {
   Props,
   Plugin,
   HtmlTags,
+  OptionValidationContext,
 } from '@docusaurus/types';
 import {Configuration, Loader} from 'webpack';
 import {generateBlogFeed, generateBlogPosts} from './blogUtils';
 
-const DEFAULT_OPTIONS: PluginOptions = {
-  path: 'blog', // Path to data on filesystem, relative to site dir.
-  routeBasePath: 'blog', // URL Route.
-  include: ['*.md', '*.mdx'], // Extensions to include.
-  postsPerPage: 10, // How many posts per page.
-  blogListComponent: '@theme/BlogListPage',
-  blogPostComponent: '@theme/BlogPostPage',
-  blogTagsListComponent: '@theme/BlogTagsListPage',
-  blogTagsPostsComponent: '@theme/BlogTagsPostsPage',
-  showReadingTime: true,
-  remarkPlugins: [],
-  rehypePlugins: [],
-  editUrl: undefined,
-  truncateMarker: /<!--\s*(truncate)\s*-->/, // Regex.
-  admonitions: {},
-};
-
-function assertFeedTypes(val: any): asserts val is FeedType {
-  if (typeof val !== 'string' && !['rss', 'atom', 'all'].includes(val)) {
-    throw new Error(
-      `Invalid feedOptions type: ${val}. It must be either 'rss', 'atom', or 'all'`,
-    );
-  }
-}
-
-const getFeedTypes = (type?: FeedType) => {
-  assertFeedTypes(type);
-  let feedTypes: ('rss' | 'atom')[] = [];
+const getFeedTypes = (type: FeedType) => {
+  let feedTypes: Array<Exclude<FeedType, 'all'>> = [];
```

Comment:
Wonder if it's not worth having 2 explicit named types here, like `FeedTypePreset` (with `all`) and `FeedType` (array)

That would also mean we have 2 types for plugin options:
- The type before schema validation/normalization, like `UserProvidedOptions`
- The normalized options

Reply:
I am thinking of moving it to validation

Reply:
we can use a Template for `UserProvidedOptions`.

Reply:
Yes, I think all transformations should rather be done ahead of time during the validation/normalization (transforming UserProvidedOptions to NormalizedPluginOption).

What do you mean by Template?

Reply:
Template type like Array<T>, but it just added normilization to it in validation it self.

Reply:
Normalized it to array